google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -137,9 +138,10 @@ public CancellableRunnable createDeployRunner(
         wrapCallbackForUsageTracking(
             callback, deploymentConfiguration, source.getFile(), targetEnvironment));
 
-    if (targetEnvironment == AppEngineEnvironment.APP_ENGINE_STANDARD) {
+    if (targetEnvironment.isStandard()
+        || (targetEnvironment.isFlexible() && AppEngineUtil.isFlexCompat(project, source))) {
```

Comment:
yea we really do. Now that you mention it - if AppEngineEnvironment wasn't thought of as the target cloud environment, but instead we thought of it as the 'framework type', then possibly my code prior to these last changes would make more sense:

3 framework types, 'standard', 'flexible', 'flex-compat' (maybe even 'standard-compat' would make more sense).